src/main/java/com/amazon/ion/impl/IonManagedWriterOptions_1_1.kt

```diff
@@ -0,0 +1,152 @@
+ *    until it has been used at least `N` times.
+ *  - A streaming-oriented implementation could keep track of the symbols that are used, and inline any symbols not
+ *    already in the symbol table. Once a top-level value is complete, some other component could inspect the list of
+ *    new symbols and emit a Local Symbol Table append with those symbols so that they can be interned use in future
+ *    top-level values.
+ *
+ * TODO: remove "sealed" when we are confident of the API.
+ */
+@SuppressFBWarnings("IC_SUPERCLASS_USES_SUBCLASS_DURING_INITIALIZATION")
+sealed interface SymbolInliningStrategy {
+    /**
+     * Represents the different kinds of usage of a symbol token.
+     */
+    enum class SymbolKind {
+        VALUE,
+        FIELD_NAME,
+        ANNOTATION,
+    }
+
+    /**
+     * Indicates whether a particular symbol text should be written inline (as opposed to writing as a SID).
+     */
+    fun shouldWriteInline(symbolKind: SymbolKind, text: String): Boolean
+
+    private object NeverInline : SymbolInliningStrategy {
+        override fun shouldWriteInline(symbolKind: SymbolKind, text: String): Boolean = false
+        override fun toString(): String = "NEVER_INLINE"
+    }
+
+    private object AlwaysInline : SymbolInliningStrategy {
+        override fun shouldWriteInline(symbolKind: SymbolKind, text: String): Boolean = true
+        override fun toString(): String = "ALWAYS_INLINE"
+    }
+
+    companion object {
+        /**
+         * A [SymbolInliningStrategy] that causes all symbols to be written as a SID,
+         * interning the text in the Local Symbol Table if necessary.
+         *
+         * This is equivalent to the behavior of symbols in Ion 1.0 binary.
+         */
+        @JvmField
+        val NEVER_INLINE: SymbolInliningStrategy = NeverInline
+
+        /**
+         * A [SymbolInliningStrategy] that causes all symbols with known text to have their text written inline.
+         */
+        @JvmField
+        val ALWAYS_INLINE: SymbolInliningStrategy = AlwaysInline
+    }
+}
+
+/**
+ * TODO: Proper documentation.
+ *
+ * See [SymbolInliningStrategy] for a similar strategy interface.
+ *
+ * TODO: remove "sealed" when we are confident of the API.
+ */
+@SuppressFBWarnings("IC_SUPERCLASS_USES_SUBCLASS_DURING_INITIALIZATION")
+sealed interface LengthPrefixStrategy {
+    /**
+     * Indicates whether a container should be written using a length prefix.
+     *
+     * TODO: See if we can add other context, such as annotations that are going to be added to this container,
+     *       the field name (if this container is in a struct), or the delimited/prefixed status of the parent
+     *       container.
+     *
+     * With more context, we could enable strategies like:
+     *   - Write lists with annotation `X` as a delimited container.
+     */
+    fun writeLengthPrefix(containerType: ContainerType, depth: Int): Boolean
```

Comment:
Might be a little more descriptive and consistent with `SymbolInliningStrategy`
```suggestion
    fun shouldWriteLengthPrefix(containerType: ContainerType, depth: Int): Boolean
```